kommunicate/src/main/java/com/applozic/mobicomkit/api/conversation/MobiComConversationService.java

```diff
@@ -442,8 +444,23 @@ public synchronized List<Message> getMessages(Long startTime, Long endTime, Cont
                         FileClientService fileClientService = new FileClientService(context);
                         fileClientService.loadContactsvCard(message);
                     }
-                    if (Message.MetaDataType.HIDDEN.getValue().equals(message.getMetaDataValueForKey(Message.MetaDataType.KEY.getValue())) || Message.MetaDataType.PUSHNOTIFICATION.getValue().equals(message.getMetaDataValueForKey(Message.MetaDataType.KEY.getValue()))) {
-                        continue;
+
+
+                    if (Message.MetaDataType.HIDDEN.getValue().equals(message.getMetaDataValueForKey(Message.MetaDataType.KEY.getValue()))
+                            || Message.MetaDataType.PUSHNOTIFICATION.getValue().equals(message.getMetaDataValueForKey(Message.MetaDataType.KEY.getValue()))
+                    ) {
+                        // Check if message is conversation summary.
+                        boolean isKmSummary;
+                        if (message.getMetadata().containsKey(KM_SUMMARY)) {
+                             isKmSummary = Boolean.parseBoolean(message.getMetaDataValueForKey(KM_SUMMARY));
```

Comment:
Optimize code, try something like: 
  boolean isKmSummary = message.getMetadata().containsKey(KM_SUMMARY) && Boolean.parseBoolean(message.getMetaDataValueForKey(KM_SUMMARY));

Reply:
Done Added in [2410150](https://github.com/Kommunicate-io/Kommunicate-Android-Chat-SDK/pull/565/commits/2410150a3bea2d2b2d2dfd0479b5012cc515f3d8)